uwblueprint/project-read-frontend
# Allow users to view, edit, and select fields

### Notion ticket link



[Toggle questions on and off in the "create form" step](https://www.notion.so/uwblueprintexecs/Sprint-3-481954865bf444498031befeafaad95a?p=4f3d4e2d38144db9944d85b7571fe8dc)
[Field component (read only) and meta meta form page](https://www.notion.so/uwblueprintexecs/Sprint-3-481954865bf444498031befeafaad95a?p=f65435571bb640b0a36c12840a1fd656)
[Edit field component](https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=f7211296e7ca48f1b1b1bb98b47dfeca)



### Implementation description

- implements a `FieldEditor` component, which displays a card with the field's data
- implements a `FormEditor` component, which displays each of the fields to show in the meta (meta) forms
  - registered a `/fields` route for the global fields setting page, and added a button in the top right menu
  - each dynamic field has an edit button that lets users rename the field, change the question type, and edit its options
- **create session page**
  - each dynamic field has a toggle - when changed, these ids get saved in the sessions state
  - no edit buttons

#### Global fields settings
<img width="493" alt="Screen Shot 2021-08-12 at 8 22 50 PM" src="https://user-images.githubusercontent.com/37346399/129286153-b627db2f-7a29-470c-95d8-8bd9b8aa8dae.png">
<img width="1536" alt="Screen Shot 2021-08-12 at 8 21 23 PM" src="https://user-images.githubusercontent.com/37346399/129286046-937eee1e-9428-41b2-bf18-41eea14e2da0.png">
<img width="1536" alt="Screen Shot 2021-08-12 at 8 22 05 PM" src="https://user-images.githubusercontent.com/37346399/129286107-1edef505-c392-4e03-bab0-57621b496aa7.png">
<img width="1536" alt="Screen Shot 2021-08-21 at 9 16 53 PM" src="https://user-images.githubusercontent.com/37346399/130338823-e1a02d40-8b38-42af-95e5-06804c6c57b1.png"> 

#### Configure session form
<img width="1536" alt="Screen Shot 2021-08-12 at 8 26 23 PM" src="https://user-images.githubusercontent.com/37346399/129286368-7854ef51-188c-41f6-b973-7b5e376ad7a2.png">
<img width="1536" alt="Screen Shot 2021-08-12 at 8 26 57 PM" src="https://user-images.githubusercontent.com/37346399/129286411-14939548-b66c-4ad1-bc68-6d1324af9128.png">



### Steps to test

1. navigate to each of the pages above and verify things look right!
2. try playing with the editors and ensure interactions feel smooth



### What should reviewers focus on?

- code quality, readability

### Checklist
